matlab/JSBSimInterface.cpp

```diff
@@ -0,0 +1,973 @@
+#include "JSBSimInterface.h"
+#include <models/FGAircraft.h>
+#include <models/FGAccelerations.h>
+#include <math/FGQuaternion.h>
+
+/* 2021-03-21 compiles with JSBSim 1.1.5
+ * Compiles with JSBSim checkout a59596f8f0d4c7b4f7ba24e99997bc2071d6cb72
+ */
+
+JSBSimInterface::JSBSimInterface(FGFDMExec *fdmex)
+{
+	_ac_model_loaded = false;
+	fdmExec = fdmex;
+	propagate = fdmExec->GetPropagate().get();
+	accel = fdmExec->GetAccelerations().get();
+	accel->InitModel();
+	auxiliary = fdmExec->GetAuxiliary().get();
+	aerodynamics = fdmExec->GetAerodynamics().get();
+	propulsion = fdmExec->GetPropulsion().get();
+	fcs = fdmExec->GetFCS().get();
+	ic = new FGInitialCondition(fdmExec);
+	//verbosityLevel = JSBSimInterface::eSilent;
+}
+JSBSimInterface::JSBSimInterface(FGFDMExec *fdmex, double dt)
+{
+  _ac_model_loaded = false;
+  fdmExec = fdmex;
+  fdmExec->Setdt(dt);
+  mexPrintf("Simulation dt set to %f\n",fdmExec->GetDeltaT());
+  propagate = fdmExec->GetPropagate().get();
+  accel = fdmExec->GetAccelerations().get();
+  accel->InitModel();
+  auxiliary = fdmExec->GetAuxiliary().get();
+  aerodynamics = fdmExec->GetAerodynamics().get();
+  propulsion = fdmExec->GetPropulsion().get();
+  fcs = fdmExec->GetFCS().get();
+  ic = new FGInitialCondition(fdmExec);
+  //verbosityLevel = JSBSimInterface::eSilent;
+}
+//%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+JSBSimInterface::~JSBSimInterface(void)
+{
+	//fdmExec = 0L;
+    //JSBSim::~FGFDMExec();
+	//delete ic;
+    delete fdmExec;
```

Comment:
What would you recommend doing here? 

A current issue is that i have to run clearSF.m after each Simulink simulation, and I think it is because the destructor isn't working properly. I am not sure how to resolve that issue however.

Reply:
What about calling `FDMExec.ResetToInitialConditions(mode)` in `mdlInitializeConditions()`?

Reply:
I think the problem lies in the variable `FDMExec` which you made a global variable in the file `JSBSim_SFunction.cpp` and as such "survives" as long as the S-Function exists; hence the need to clear everything. Given that `FDMExec` is only used by the class `JSBSimInterface` I would suggest a different approach and make the variable `FDMExec` a member of the `JSBSimInterface` class.

You also need to delete the instance of `JSBSimInterface` in `mdlTerminate`. At the moment, since the instance is not deleted your SFunction is leaking memory.

Reply:
Once you have made `FGFDMExec` a member of `JSBSimInterface`, you can call `delete` on `fdmExec`